domain-management/src/test/java/org/jboss/as/domain/management/security/PropertiesFileLoaderTestCase.java

```diff
@@ -0,0 +1,180 @@
+        if (tmpFile != null)
+            tmpFile.delete();
+    }
+
+    private void writeTestDataToFile(final File tmpFile) throws Exception {
+        FileOutputStream fos = new FileOutputStream(tmpFile);
+        for (String s : properties) {
+            fos.write(s.getBytes());
+            fos.write('\n');
+        }
+        fos.flush();
+        fos.close();
+    }
+
+    private void verifyProperties(final Properties props, final int expectedSize) {
+        verifyProperties(props, expectedSize, null, null);
+    }
+
+    private void verifyProperties(final Properties props, final int expectedSize, final String changedKey, final String expectedValue) {
+        for (String s : properties) {
+            if (s.startsWith("#")) // skip comments, the persist methods on PropertiesFileLoader will re-read those and write them back out
+                continue;
+            String[] parts = s.split("=");
+            String user = parts[0];
+            String password = parts[1];
+            if (changedKey != null && changedKey.equals(user)) {
+                if (expectedValue == null) {
+                    Assert.assertNull(props.get(changedKey));
+                } else {
+                    Assert.assertEquals(expectedValue, props.get(changedKey));
+                }
+            } else {
+                Assert.assertNotNull(props.getProperty(user));
+                Assert.assertEquals(password, props.getProperty(user));
+            }
+        }
+        // -1 because we don't count the comment
+        Assert.assertEquals(properties.length-1, expectedSize);
+    }
+
+    @Test
+    public void testLoad() throws Exception {
+        File tmpFile = null;
+        try {
+            tmpFile = createTempFile();
+            writeTestDataToFile(tmpFile);
+
+            PropertiesFileLoader loader = new PropertiesFileLoader(tmpFile.getAbsolutePath(), null);
+            loader.start(null);
+            Properties props = loader.getProperties();
+            verifyProperties(props, props.size());
+            loader.stop(null);
+        } finally {
+            cleanupTempFile(tmpFile);
+        }
+    }
+
+    @Test
+    public void testAdd() throws Exception {
+        File tmpFile = null;
+        try {
+            tmpFile = createTempFile();
+            writeTestDataToFile(tmpFile);
+
+            PropertiesFileLoader loader = new PropertiesFileLoader(tmpFile.getAbsolutePath(), null);
+            loader.start(null);
+            Properties props = loader.getProperties();
+            props.put("NEW", "VALUE");
+            loader.persistProperties();
+            loader.stop(null);
+
+            // reload the file and make sure everything is there
+            PropertiesFileLoader loader2 = new PropertiesFileLoader(tmpFile.getAbsolutePath(), null);
+            loader2.start(null);
+            Properties props2 = loader2.getProperties();
+            verifyProperties(props2, props2.size()-1, "NEW", "VALUE");
```

Comment:
This call doesn't validate that NEW is present with a value of VALUE, since that's not an entry in the  "properties" array. You could pass null for both params and get the same effect.

Kind of a nit; it just confused me a bit.